crates/ibc/abci-tag/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+use quote::quote;
+
+use proc_macro::TokenStream;
+use proc_macro2::Ident;
+use syn::{parse_macro_input, DataStruct, DeriveInput, Ident as SynIdent};
+
+#[proc_macro_attribute]
+pub fn abci_tag(args: TokenStream, item: TokenStream) -> TokenStream {
```

Comment:
```suggestion
pub fn abci_event_attr_key(args: TokenStream, item: TokenStream) -> TokenStream {
```

Could please change the name? I find this more descriptive.